R/FilterStateChoices.R

```diff
@@ -448,63 +448,92 @@ ChoicesFilterState <- R6::R6Class( # nolint
               NULL
             }
 
-            if (private$is_checkboxgroup()) {
-              updateCountBars(
-                inputId = "labels",
-                choices = private$get_choices(),
-                countsmax = private$choices_counts,
-                countsnow = countsnow
-              )
-            } else {
-              labels <- mapply(
-                FUN = make_count_text,
-                label = private$get_choices(),
-                countnow = if (is.null(countsnow)) rep(list(NULL), length(private$get_choices())) else countsnow,
-                countmax = private$choices_counts
-              )
-              teal.widgets::updateOptionalSelectInput(
-                session = session,
-                inputId = "selection",
-                choices = stats::setNames(private$get_choices(), labels),
-                selected = private$get_selected()
-              )
-            }
-            NULL
-          })
-
-          private$observers$selection <- observeEvent(
-            ignoreNULL = FALSE,
-            ignoreInit = TRUE, # ignoreInit: should not matter because we set the UI with the desired initial state
-            eventExpr = input$selection,
-            handlerExpr = {
-              logger::log_trace("ChoicesFilterState$server_inputs@2 changed selection, id: { private$get_id() }")
-
-              selection <- if (is.null(input$selection) && private$is_multiple()) {
-                character(0)
-              } else if (isTRUE(length(input$selection) != 1) && !private$is_multiple()) {
-                showNotification(paste(
-                  "This filter exclusively supports single selection.",
-                  "Any additional choices made will be disregarded."
-                ))
+            # update should be based on a change of counts only
+            shiny::isolate({
```

Comment:
@chlebowa @m7pr I made this change because labels should change only when the reactive filtered data got changed